File: core/src/io/github/jerukan/game/board/Tile.java
package io.github.jerukan.game.board;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.Pixmap;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.Sprite;
import io.github.jerukan.util.Assets;
import io.github.jerukan.util.Constants;
import io.github.jerukan.util.Position;

public class Tile {

    private Position position;
    private Sprite sprite;
    private Texture rect;

    private float height;
    private Terrain.TerrainInfo terrain;

    public Tile(Position position, float height) {
        this.position = position;
        this.height = height;
        terrain = Terrain.terrainInfoFromHeight(height);
        sprite = new Sprite(terrain.texture);
        sprite.setSize(Constants.TILE_SIZE, Constants.TILE_SIZE);
        Pixmap pixmap = new Pixmap((int)(sprite.getWidth()), (int)(sprite.getHeight()), Pixmap.Format.RGBA8888);
        pixmap.setColor(Color.BLACK);
        pixmap.drawRectangle(0, (int)(sprite.getY()), pixmap.getWidth(), pixmap.getHeight());
        rect = new Texture(pixmap);
        pixmap.dispose();
    }

    public void render(Batch batch) {
        batch.draw(sprite.getTexture(), sprite.getX(), sprite.getY(), sprite.getWidth(), sprite.getHeight());
        batch.draw(rect, sprite.getX(), sprite.getY());
    }

    public void setSpritePosition(float x, float y) {
        sprite.setPosition(x, y);
    }

    public Sprite getSprite() {
        return sprite;
    }

    public float getHeight() {
        return height;
    }

    public int getSpeedConsump() {
        return terrain.speedConsump;
    }

    public void dispose() {
        rect.dispose();
    }
}

File: core/src/io/github/jerukan/game/board/Tile.kt
package io.github.jerukan.game.board

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.Pixmap
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.Sprite
import io.github.jerukan.util.Assets
import io.github.jerukan.util.Constants
import io.github.jerukan.util.Position

class Tile(private val position: Position, val height: Float) {
    val sprite: Sprite
    private val rect: Texture
    val terrain: Terrain.TerrainInfo

    val speedConsump: Int
        get() = terrain.speedConsump

    init {
        terrain = Terrain.terrainInfoFromHeight(height)
        sprite = Sprite(terrain.texture)
        sprite.setSize(Constants.TILE_SIZE.toFloat(), Constants.TILE_SIZE.toFloat())
        val pixmap = Pixmap(sprite.width.toInt(), sprite.height.toInt(), Pixmap.Format.RGBA8888)
        pixmap.setColor(Color.BLACK)
        pixmap.drawRectangle(0, sprite.y.toInt(), pixmap.width, pixmap.height)
        rect = Texture(pixmap)
        pixmap.dispose()
    }

    fun render(batch: Batch) {
        batch.draw(sprite.texture, sprite.x, sprite.y, sprite.width, sprite.height)
        batch.draw(rect, sprite.x, sprite.y)
    }

    fun setSpritePosition(x: Float, y: Float) {
        sprite.setPosition(x, y)
    }

    fun dispose() {
        rect.dispose()
    }
}